### PR TITLE
wkhtmltopdf: unbreak package

### PR DESCRIPTION
**Pull request by zimbatm**
## Description of changes

Some strategic changes to allow the package to be built again.

Motivation: Odoo depends on this package.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by zimbatm** [comment]
Result of `nixpkgs-review pr 327641` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>odoo</li>
    <li>odoo.dist</li>
    <li>odoo16</li>
    <li>odoo16.dist</li>
    <li>wkhtmltopdf</li>
  </ul>
</details>

File: pkgs/tools/graphics/wkhtmltopdf-bin/default.nix [hunk 1]
```diff
@@ -57,16 +58,24 @@ let
...
+      (lib.getLib libjpeg)
       (lib.getLib libpng)
     ];
 
-    unpackPhase = "tar -xf $src";
+    unpackPhase = ''
+      runHook preUnpack
+
+      mkdir pkg
+      dpkg-deb -x $src pkg
```
Comment (Aleksanaa) [inline pkgs/tools/graphics/wkhtmltopdf-bin/default.nix:69]: Does fetchzip decompress it? idk

Comment (zimbatm) [reply pkgs/tools/graphics/wkhtmltopdf-bin/default.nix:69]: Technally yes, there is one package doing it, pkgs/by-name/te/tetrio-desktop/package.nix. All of the others are using fetchurl.

I prefer using fetchurl as the hash is on the content and can be easily calculated separately.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -14433,6 +14433,7 @@ with pkgs;
   wireguard-go = callPackage ../tools/networking/wireguard-go { };
 
   wkhtmltopdf-bin = callPackage ../tools/graphics/wkhtmltopdf-bin { };
+  wkhtmltopdf = wkhtmltopdf-bin;
```
Comment (zimbatm) [inline pkgs/top-level/all-packages.nix:14436]: hydra doesn't allow aliases to be passed to packages.

we could rename all the inputs, or keep this manual alias here.

**Review by kalbasit — APPROVED** [review]

**Review by drupol — APPROVED** [review]
Odoo using Nix? Interesting :)

**Comment by zimbatm** [comment]
yes, I have some follow-up PRs and a blog post coming up. Definitely not production ready.

**Comment by drupol** [comment]
@zimbatm there's this that might help? https://github.com/novacode-nl/nix-odoo

**Comment by zimbatm** [comment]
I've seen it; it's more for a dev setup than to deploy Odoo.

File: pkgs/top-level/all-packages.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/top-level/all-packages.nix:14436]: #327890

**Pull request merged**